node-webservice/react-client/src/Components/GameplayPage.jsx

```diff
@@ -244,34 +252,119 @@ class GameplayPage extends Component {
           }
           copy.push(newPart);
           newPart.direction = this.rotate(newPart.direction, x, y);
-          this.setState({ parts: copy});
-
-        
-  
+          this.setState({ parts: copy });
+        }
       }
-
-      
     }
-
-
   }
+  
   PartExistsIn(arr, target) {
     var found = 0;
     arr.forEach(element => {
-      if (element[0] === target[0] && element[1] === target[1] ) {
+      if (element[0] === target[0] && element[1] === target[1]) {
         found = 1;
       }
     })
     return found;
   }
 
+  deleteInvalid(partsCopy){
```

Comment:
The functionality for this stuff seems to work well, but imo the code for handleCellClicked() and deleteInvalid() have become very dense and hard to read. To help keep it maintainable it would be good to work on the readability and to add some tests
This could defs be a later task